### DISCUSSION TITLE
BOINC Integration

### DISCUSSION THREAD

**Discussion by AenBleidd** [official-stockfish/Stockfish#5043] [Ideas]
Hello folks,

Is there any interest of bringing this project to the BOINC infrastructure that is running on thousands of volunteer devices?
You might heard about some of the popular projects like SETI@home, World Community Grid, Einstein@home, etc.
You can check the details here: https://boinc.berkeley.edu/

If you would like to explore this opportunity, I'll be happy to make an introduction to it.

Don't hesitate to contact either me via email (lestat.de.lionkur@gmail.com) or the Director of BOINC David P. Anderson: https://boinc.berkeley.edu/anderson/

**Comment by jshriver** [comment]
Good timing as I'm interested. I currently run the openchessdb.org
website.  Just 2 days ago I began researching the possibility of moving it
to BOINC as a means to gain exposure and save time not re-implementing
basic functions. (user accounts, work unit rewards, metrics, etc).  Willing
to help on another project if stockfish wanted to use it for other means
such as NN-testing.

-Josh

**Reply by AenBleidd** [reply]
Hello @jshriver, thank you for your interest.
If you want - we can make a call with you to discuss your project and possible BOINC Integration.
Feel free to contact me via email with your availability.

**Comment by Disservin** [comment]
Hi @AenBleidd, 

Thanks for approaching us with this :D 
This does depend a bit on how a BOINC integration would look like... unfortunately I didn't find a clear explanation how projects are added to BOINC. Can you point me to some good information on this?

In our current setup we have a python worker script (which has it's own requirements), which sends back the results to our server.

**Reply by AenBleidd** [reply]
Hello @Disservin, thank you for your interest.
If you want - we can make a call, and I'll explain available options for you and we can then together find the best solution for you.
I have left you my email, so feel free to contact me there with your availability.
In the meantime, you can read some documentation here: https://github.com/BOINC/boinc/wiki/Computing-with-boinc

**Reply by MrBrain295** [reply]
Would this just be like the Moo! wrapper but for Fishtest?

**Reply by Disservin** [reply]
dunno what moo! is..

but i forgot to post an update here:

I had a talk with Vitalii (a long time ago now) about BOINC and its capabilities. Unfortunately I currently think the architectural changes which this would require are not possible to implement at the moment.
We might want to revisit this in the future though.

**Reply by MrBrain295** [reply]
[moo!](https://www.distributed.net/Main_Page) is a distributed computing platform that has a [boinc wrapper](https://moowrap.net/).
